lib/core/stores/local.ts

```diff
@@ -73,7 +76,7 @@ export class LocalPromiseStore implements IPromiseStore {
     data: string | undefined,
   ): Promise<ResolvedPromise | RejectedPromise | CanceledPromise | TimedoutPromise> {
     return this.storage.rmw(id, (promise) => {
-      if (promise) {
+      if (isDurablePromise(promise)) {
```

Comment:
Can we revert these checks? Because of the rmw type (DurablePromise | undefined) the only type guard we should require is that promise is not undefined.